broker/httpd/auth.go

```diff
@@ -53,15 +77,17 @@ func (s *Server) loginHandler(w http.ResponseWriter, r *http.Request) {
 	s.authCookie[userCookie.Value] = Cookieinfo
 	s.cookieMutex.Unlock()
 
-	http.Redirect(w, r, "/", http.StatusFound)
+	destinationPath := getRedirDestination(r)
+
+	http.Redirect(w, r, destinationPath, http.StatusFound)
```

Comment:
Please collapse this into a single line:
`http.Redirect(w, r, getRedirDestination(r), http.StatusFound)`